lib/quicly.c

```diff
@@ -3869,8 +3869,7 @@ static int do_send(quicly_conn_t *conn, quicly_send_context_t *s)
     if (conn->application != NULL && (s->current.cipher = &conn->application->cipher.egress.key)->header_protection != NULL) {
         s->current.first_byte = conn->application->one_rtt_writable ? QUICLY_QUIC_BIT : QUICLY_PACKET_TYPE_0RTT;
         /* acks */
-        if (conn->application->one_rtt_writable && conn->egress.send_ack_at <= conn->stash.now &&
-            conn->application->super.unacked_count != 0) {
+        if (conn->application->one_rtt_writable && conn->application->super.unacked_count != 0) {
```

Comment:
@janaiyengar This is a behavior change that I would like to discuss.

Up until now, we have been withholding ACKs until either the delayed ack timer fires or until unacked_count reaches the threshold. Is that the thing we should do?

At the moment, this PR changes the behavior to send ACK whenever there is new information, merely because doing so does not require change in the test cases that we already have. Number of packets being exchanged prior to the transaction test has changed - causing behavioral changes in the transaction test. We can fix the transaction tests, but would like to discuss what we should be doing here before making such change.